src/shared/specifier.c

```diff
@@ -208,6 +208,23 @@ int specifier_short_host_name(char specifier, const void *data, const char *root
         return 0;
 }
 
+int specifier_pretty_host_name(char specifier, const void *data, const char *root, const void *userdata, char **ret) {
+        char *n = NULL;
+        int r;
+
+        assert(ret);
+
+        r = get_pretty_hostname(&n);
+        if (r < 0) {
+                n = strdup("(none)");
```

Comment:
Not sure, but if the pretty hostname is not set, then it is better to fallback the short hostname??

Also, you can drop the assignment to `r`.